linkerd/meshtls/boring/src/client.rs

```diff
@@ -0,0 +1,163 @@
+use boring::ssl;
+use linkerd_error::Result;
+use linkerd_identity::Name;
+use linkerd_io as io;
+use linkerd_stack::{NewService, Service};
+use linkerd_tls::{
+    client::AlpnProtocols, ClientTls, HasNegotiatedProtocol, NegotiatedProtocolRef, ServerId,
+};
+use std::{future::Future, pin::Pin};
+use tokio::sync::watch;
+
+#[derive(Clone)]
+pub struct NewClient(watch::Receiver<ssl::SslConnector>);
+
+#[derive(Clone)]
+pub struct Connect {
+    server_id: Name,
+    connector: ssl::SslConnector,
+}
+
+pub type ConnectFuture<I> = Pin<Box<dyn Future<Output = io::Result<ClientIo<I>>> + Send>>;
+
+#[derive(Debug)]
+pub struct ClientIo<I>(tokio_boring::SslStream<I>);
+
+// === impl NewClient ===
+
+impl NewClient {
+    pub(crate) fn new(rx: watch::Receiver<ssl::SslConnector>) -> Self {
+        Self(rx)
+    }
+}
+
+impl NewService<ClientTls> for NewClient {
+    type Service = Connect;
+
+    fn new_service(&self, target: ClientTls) -> Self::Service {
+        Connect::new(target, (*self.0.borrow()).clone())
+    }
+}
+
+impl std::fmt::Debug for NewClient {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("NewClient").finish()
+    }
+}
+
+// === impl Connect ===
+
+impl Connect {
+    pub(crate) fn new(client_tls: ClientTls, connector: ssl::SslConnector) -> Self {
+        let ServerId(server_id) = client_tls.server_id;
+
+        if let Some(AlpnProtocols(protocols)) = client_tls.alpn {
+            if !protocols.is_empty() {
+                todo!("support ALPN")
+            }
+        }
+
+        Self {
+            server_id,
+            connector,
+        }
+    }
+}
+
+impl<I> Service<I> for Connect
+where
+    I: io::AsyncRead + io::AsyncWrite + Send + Sync + Unpin + 'static,
+{
+    type Response = ClientIo<I>;
+    type Error = io::Error;
+    type Future = ConnectFuture<I>;
+
+    fn poll_ready(
+        &mut self,
+        _cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Result<(), Self::Error>> {
+        std::task::Poll::Ready(Ok(()))
```

Comment:
nit, take it or leave it: maybe worth importing `Poll` and `Context` in this module?